app/containers/HomeContainer/tests/index.test.js

```diff
@@ -46,4 +57,61 @@ describe('<HomeContainer /> tests', () => {
     await timeout(500);
     expect(submitSpy).toBeCalled();
   });
+  it('should dispatch githubRepos when repoName is passed', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchGithubRepos({ repoName: 'Mac' });
+    expect(dispatch).toHaveBeenCalled();
+  });
+  it('should dispatch clear Repos when repoName is empty', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchClearGithubRepos();
+    expect(dispatch.mock.calls[0][0]).toEqual({ type: homeContainerTypes.CLEAR_GITHUB_REPOS });
+  });
+
+  it('should trigger dispatchGithubRepos on mount', async () => {
+    const getGithubReposSpy = jest.fn();
+    renderProvider(<HomeContainer dispatchGithubRepos={getGithubReposSpy} repoName="mac" />);
+    await timeout(500);
+    expect(getGithubReposSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should show repoError ', () => {
+    const { getByText } = renderProvider(<HomeContainer reposError="error" />);
+    expect(getByText('error')).toBeTruthy();
+  });
+
+  it('should set loading to true when repoName present', async () => {
+    const getGithubReposSpy = jest.fn();
+    renderProvider(<HomeContainer repoName="mac" dispatchGithubRepos={getGithubReposSpy} />);
+    await timeout(500);
+    expect(getGithubReposSpy).toHaveBeenCalledTimes(1);
+    expect(setState).toBeCalledWith(true);
```

Comment:
this will change based on what we spoke about earlier. Don't mock react.useState